### DISCUSSION TITLE
How to translate some templates like `__(foo + 'bar')`

### DISCUSSION THREAD

**Discussion by liuliangsir** [privatenumber/webpack-localize-assets-plugin#45] [Q&A] [answered]
As the above title says, I would like to translate some strings, which contain a certain variables that might be from `import` statement. The following is a code snippet.

```
import foo from 'foo'

const bar = __(foo + 'baz')
```

**Comment by liuliangsir** [comment]
@privatenumber cc

**Comment by privatenumber** [comment, marked as answer]
It's not possible with this plugin because it only does basic static analysis.

As a work around, you can create a map:
```ts
import foo from 'foo'

const mapping = {
	'a-baz': __('key-1'),
	'b-baz': __('key-2')
}

console.log(mapping[foo + '-baz']);
```
